Scripts/script-CommonServerPython.yml

```
@@ -1223,6 +1223,89 @@ script: |-
+        :rtype: ``(datetime.datetime, datetime.datetime)`` or ``(int, int)`` or ``(str, str)``
+      """
+      range_split = date_range.split(' ')
+      if len(range_split) != 2:
+          return_error('date_range must be "number date_range_unit", examples: (2 hours, 4 minutes, 6 month, 1 day, etc.)')
+
+      number = int(range_split[0])
+      if not range_split[1] in ['minute', 'minutes', 'hour', 'hours', 'day', 'days', 'month', 'months', 'year', 'years']:
+          return_error('The unit of date_range is invalid. Must be minutes, hours, days, months or years')
+
+      end_time = datetime.now()
+      start_time = datetime.now()
+      unit = range_split[1]
+      if 'minute' in unit:
+          start_time = end_time - timedelta(minutes=number)
+      elif 'hour' in unit:
+          start_time = end_time - timedelta(hours=number)
+      elif 'day' in unit:
+          start_time = end_time - timedelta(days=number)
+      elif 'month' in unit:
+          start_time = end_time - timedelta(days=number*30)
+      elif 'year' in unit:
+          start_time = end_time - timedelta(days=number*365)
+
+      if to_timestamp:
+          return date_to_timestamp(start_time), date_to_timestamp(end_time)
+
+      if date_format:
+          return datetime.strftime(start_time, date_format), datetime.strftime(end_time, date_format)
+
+      return start_time, end_time
+
+  def timestamp_to_datestring(timestamp, date_format="%Y-%m-%dT%H:%M:%S.000Z"):
+      """
+        Parses timestamp (milliseconds) to a date string in the provided date format (by default: ISO 8601 format)
+        Examples: (1541494441222, 1541495441000, etc.)
+
+        :type timestamp: ``int`` or ``str``
+        :param timestamp: The timestamp to be parsed (required)
+
+        :type date_format: ``str``
+        :param date_format: The date format the timestamp should be parsed to. (optional)
+
+        :return: The parsed timestamp in the date_format
+        :rtype: ``str``
+      """
+      return datetime.fromtimestamp(int(timestamp) / 1000.0).strftime(date_format)
+
+  def date_to_timestamp(date_str_or_dt, date_format='%Y-%m-%dT%H:%M:%S'):
+      """
+        Parses date_str_or_dt in the given format (default: %Y-%m-%dT%H:%M:%S) to miliseconds
+        Examples: ('2018-11-06T08:56:41', '2018-11-06T08:56:41', etc.)
+
+        :type date_str_or_dt: ``str`` or ``datetime.datetime``
+        :param date_str_or_dt: The date to be parsed. (required)
+
+        :type date_format: ``str``
+        :param date_format: The date format of the date string (will be ignored if date_str_or_dt is of type datetime.datetime). (optional)
+
+        :return: The parsed timestamp.
+        :rtype: ``int``
+      """
+      if isinstance(date_str_or_dt, basestring):
+          return int(time.mktime(time.strptime(date_str_or_dt, date_format))*1000)
+
+      # otherwise datetime.datetime
+      return int(time.mktime(time.strptime(datetime.strftime(date_str_or_dt, date_format), date_format))*1000)
```

Converting a datetime object like this may cause issues, it is better not to convert it to string and parse it again.
```suggestion
      return int(time.mktime(date_str_or_dt.timetuple()*1000))
```